console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -29,19 +32,47 @@ export const MetaDataRow: React.FC<IMetaDataRow> = ({
   setMetadataList,
   rowIndex
 }) => {
-  const metadataRow = metadataList[rowIndex];
+  const currentRow = metadataList[rowIndex];
   const [validationStatus, setValidationStatus] = useState<
     ValidationStatusType.DEFAULT | ValidationStatusType.ERROR
   >(ValidationStatusType.DEFAULT);
 
-  const onSelectType = (typeValue: string) => {
-    const validationStatus = getValidationStatus(typeValue, metadataRow.value);
+  const updateMetadataList = (property: string, value: string) => {
+    let updatedMetadataList = [...metadataList];
+    switch (property) {
+      case "key":
+        updatedMetadataList[rowIndex].key = value;
+        break;
+      case "type":
+        updatedMetadataList[rowIndex].type = value;
+        if (isObjectOrArray(value as any))
+          updatedMetadataList[rowIndex].value = "";
+        break;
+      case "value":
+        updatedMetadataList[rowIndex].value = value;
+        break;
+    }
+    setMetadataList(updatedMetadataList);
+  };
+
+  const handleTypeChange = (type: string) => {
+    const validationStatus = getValidationStatus(type, currentRow.value);
     setValidationStatus(validationStatus);
-    let updatedTypeMetadata = [...metadataList];
-    updatedTypeMetadata[rowIndex].type = typeValue;
-    if (isObjectOrArray(typeValue as any))
-      updatedTypeMetadata[rowIndex].value = "";
-    setMetadataList(updatedTypeMetadata);
+    updateMetadataList("type", type);
+  };
+
+  const handlePropertyChange = (property: string) => {
+    updateMetadataList("key", property);
+  };
+
+  const handleValueChange = (value: string, _: any) => {
+    const validationStatus = getValidationStatus(currentRow.type, value);
+    setValidationStatus(validationStatus);
+    updateMetadataList("value", value);
+  };
+
+  const isObjectOrArray = (type: DataType.ARRAY | DataType.OBJECT) => {
```

Comment:
it would be better to move this function in top level utils so that it can be reuse.

Reply:
Agreed.